src/routing.js

```diff
@@ -107,26 +104,27 @@ export function getAppPath({
     return `/${action}`
   }
 
-  if (!fullDao) {
-    return `/${search}`
+  // Only keep the DAO name if it ends in aragonid.eth
+  if (dao.endsWith(ARAGONID_ENS_DOMAIN)) {
+    dao = dao.substr(0, dao.indexOf(ARAGONID_ENS_DOMAIN) - 1)
   }
 
-  const dao =
-    fullDao.indexOf(ARAGONID_ENS_DOMAIN) > -1
-      ? fullDao.substr(0, fullDao.indexOf(ARAGONID_ENS_DOMAIN) - 1)
-      : fullDao
-
-  // The search takes priority over app params for now. App params are going to
-  // be replaced soon so it shouldn’t be an issue.
-  if (!search && params) {
-    search = `?p=${encodeURIComponent(params)}`
-  }
-
-  if (staticApps.has(instanceId)) {
-    return '/' + dao + staticApps.get(instanceId).route + localPath + search
+  if (!dao) {
+    return `/${search}`
   }
 
-  return `/${dao}/${instanceId}${search}`
+  // Either the address of an app instance or the path of an internal app.
+  const instancePart = staticApps.has(instanceId)
+    ? staticApps.get(instanceId).route
+    : `/${instanceId}`
+
+  return (
+    '/' +
+    dao +
+    instancePart +
+    addStartingSlash(encodeAppPath(instancePath)) +
```

Comment:
We've already ensured that `instancePath` starts with a string. What do you think about lifting the `addStartingSlash` to `encodeAppPath()` so that the responsibility of forming a correct path is put just into that function?

Reply:
Yes good idea!